### PR TITLE
Disable Cypress parallelization on forks

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

This PR removes parallelization for Cypress E2E tests for fork PRs from contributors. Why? Because parallelization requires the cypress.io secret token to be passed on the action, which for security we don't allow (secrets are not passed to fork PRs). This was causing the entire test suite to be run 7 times, which aside from taking a long time and resources, was causing a lot of false negatives due to flaky tests.

In addition this PR:
- Fixes Chromatic storybook previews
- Changes parallelization for internal PRs from 7 to 4, as we were only running 1-2 files per container and we're greatly exceeding our test allocation in cypress.io.

Out of the scope of this PR:
- This PR does not fix the problem of some flaky tests still on the suite.
- Parallelization is still effectively not working for internal PRs as we've exceeded the billing capacity. This should be available once the new billing period starts.

## How did you test this code?

- Workflow running for an internal PR in this PR (and tests passing).
- Workflow running for a contributor's PR in #6793.

### CODE REVIEW

**Comment by macobo** [comment]
Note: You might need to mess with "required branches" after merging this similar to https://posthog.slack.com/archives/C0113360FFV/p1635930130151600 to avoid it being marked as in-progress forever.

Or at the very least, make not all of them required :)

File: .github/workflows/e2e_matrix.json
Comment (pauldambra) [inline .github/workflows/e2e_matrix.json:1]: what do we gain from processing this json file each time instead of having the value hardcoded in the action?

**Review by mariusandra — APPROVED** [review]
LGTM, though one cypress test still takes substantially longer than all the others

File: cypress/integration/invitesMembers.js
Comment (pauldambra) [inline cypress/integration/invitesMembers.js:98]: Should this change be in a separate PR? It's (I guess) not related to this change...

**Review by pauldambra — APPROVED** [review]
Looks like it needs the change that @macobo mentioned

But, once that's done, I need the storybook fix so LGTM :)

File: cypress/integration/invitesMembers.js
Comment (paolodamico) [reply cypress/integration/invitesMembers.js:98]: Actually this was a flaky test, that's why I included it here, sorry forgot the context

**Comment by paolodamico** [comment]
Thanks @pauldambra & @mariusandra for the review! Indeed parallelization is not fixed due to the billing stuff on cypress.io (as discussed on Slack let's wait to see what happens with that or think about our own parallelization, could also help for contributor's PRs). Updating required checks now.

File: .github/workflows/e2e_matrix.json
Comment (paolodamico) [reply .github/workflows/e2e_matrix.json:1]: Nothing really, a bit more clarity in the file, easier to read

**Pull request merged**
